### PR TITLE
New Settings language selection logic

### PR DESCRIPTION
**Pull request by impeeza**
# docs\en\05+5-translation.md
Reverting to last one, because now `fw\application\src\app\settings\scene\settings_scene_language.c` is independent to add/remove/modify language definitions.

# fw\application\src\app\settings\scene\settings_scene_language.c
Remove the need of manually add languages here, most of the code changed to automatized one.

# Language Order
Due to the automation of `fw\application\src\app\settings\scene\settings_scene_language.c` now the order of the languages on the settings menu selection are dependent to the order of the definitions of the variable on the `fw\application\src\i18n\language.c` and `fw\application\src\i18n\language.h` files.

### CODE REVIEW

**Comment by impeeza** [comment]
@solosky Thank you so much for the challenge to modify the code of settings_scene_language.c to use a `for` function to the generation of Language selection options.

To me a `C` ignorant was fun to learn the basics to be able to get a working code.

I hope the proposed code is relatively clean and easy to understand, I don put commentaries because I really don't know your style and do not want to be pushing or rude.

Please review the code and let me to know what do you think about it.

By the way took me 100 versions of the file to get the proposed one 🤣, I know I am a bad programmer 👅; I started totally blind about the `C` code but now I know a little bit, Thanks a lot.

Basically my approach was using the global `language` definition on `language.h` and `language.c` and using a simple for use the `language` variable def to populate the settings language selection sub-menu. Then use the same values for the change of language.

Tested selecting all languages and changing them, even returning from the selection with the back menu to not change the language, and worked fine to me.

**Comment by impeeza** [comment]
I don't know if the order change of the definitions of language variables on the `language.h` and `language.c` files could generate some confusion on existing devices, because I think if someone have the language `1` (English) on their `Settings.bin` file once update to the new firmware with the new order, the language becomes (繁體中文(臺灣)), what do you think?

**Comment by solosky** [comment]
Hi, thank your contribution on this PR. It looks fatastic, and implemented what did I suggested!
I am very appricate your learning jonery on C language and you finally did it!!

Regarding your order issue, the short anwser is your are correct, change the enum defination order will cause wrong language display if update to new firmware. 

So, please revert the language order change to avoid that issue.

File: fw/application/src/i18n/language.h [hunk 1]
```diff
@@ -12,20 +12,20 @@
 
 typedef enum {
     LANGUAGE_ZH_HANS,
-    LANGUAGE_EN_US,
     LANGUAGE_ZH_TW,
+    LANGUAGE_EN_US,
```
Comment (solosky) [inline fw/application/src/i18n/language.h:16]: please revert the order change, thanks

File: fw/application/src/app/settings/scene/settings_scene_language.c [hunk 2]
```diff
@@ -7,59 +7,29 @@
...
 void settings_scene_language_on_enter(void *user_data) {
 
     app_settings_t *app = user_data;
-    mui_list_view_add_item(app->p_list_view, 0xe105, "简体中文", (void *)SETTINGS_LANGUAGE_ZH_HANS);
-    mui_list_view_add_item(app->p_list_view, 0xe105, "繁體中文(臺灣)", (void *)SETTINGS_LANGUAGE_ZH_TW);
-    mui_list_view_add_item(app->p_list_view, 0xe105, "English", (void *)SETTINGS_LANGUAGE_EN_US);
-    mui_list_view_add_item(app->p_list_view, 0xe105, getLangDesc(LANGUAGE_ES_ES), (void *)SETTINGS_LANGUAGE_ES_ES);
-    mui_list_view_add_item(app->p_list_view, 0xe105, getLangDesc(LANGUAGE_HU_HU), (void *)SETTINGS_LANGUAGE_HU_HU);
-    mui_list_view_add_item(app->p_list_view, 0xe069, getLangString(_L_BACK), (void *)SETTINGS_LANGUAGE_MENU_EXIT);
+    for (SetLangOpt = 0; SetLangOpt < LANGUAGE_COUNT; SetLangOpt++){
```
Comment (solosky) [inline fw/application/src/app/settings/scene/settings_scene_language.c:29]: `SetlangOpt` can be simple to `i` as c program style. 
And dont forget to declare the variable type, e.g. `for(uint8_t i = 0; i<LANGUAGE_COUNT; i++){`

**Review by solosky — CHANGES_REQUESTED** [review]

File: fw/application/src/i18n/language.h (hunk 1 above)
Comment (impeeza) [reply fw/application/src/i18n/language.h:16]: Yeah, the order will impact the existing settings.bin, good catch.

File: fw/application/src/app/settings/scene/settings_scene_language.c (hunk 2 above)
Comment (impeeza) [reply fw/application/src/app/settings/scene/settings_scene_language.c:29]: Yeah at first I did used `i` but want to be more specific `i` is ok.

I put the definition outside of the `for` because the last item `BACK` need it to be the last one and I used the last value of `i` but also can we use the `LANGUAGE_COUNT` definition. what do you think?

Comment (solosky) [reply fw/application/src/app/settings/scene/settings_scene_language.c:29]: The user data of last `BACK` is not used, you can set macro `NULL_USER_DATA` instead.

Comment (impeeza) [reply fw/application/src/app/settings/scene/settings_scene_language.c:29]: excellent working on that!

**Comment by impeeza** [comment]
Created the https://github.com/solosky/pixl.js/pull/176 hope this work fine for you.

closing this PR
